### PR TITLE
Youtube media publishers now appear properly in Rewards panel

### PR DESCRIPTION
**Pull request by jasonrsadler**
Resolves: brave/brave-browser#1282

Ledger implementation: brave-intl/bat-native-ledger#151
UI implementation: brave/brave-ui#225

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:

1. Start Brave with clean profile and enable Rewards
2. View Youtube video for required amount of time.
3. Tab to or open Rewards page and ensure Youtube publisher is added to auto-contribute table.
4. Tab or switch back to youtube video and open Rewards panel
5. Ensure that Youtube channel publisher and icon show instead of Youtube.

*

1. Start Brave with clean profile and enable Rewards
2. View Youtube live stream for required amount of time.
3. Tab to or open Rewards page and ensure Youtube publisher is added to auto-contribute table.
4. Tab or switch back to youtube live stream and open Rewards panel
5. Ensure that Youtube channel publisher and icon show instead of Youtube.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by jasonrsadler** [comment]
Blocked on pending review of brave-int/bat-native-ledger#151

File: components/brave_rewards/extension/brave_rewards/components/panel.tsx [hunk 1]
```diff
@@ -204,7 +204,9 @@ export class Panel extends React.Component<Props, State> {
     const publisher: RewardsExtension.Publisher | undefined = this.getPublisher()
     const converted = utils.convertBalance(balance.toString(), rates)
     const notification = this.getNotification()
-
+    if (publisher && publisher.provider && publisher.provider.toLowerCase() === 'youtube') {
```
Comment (NejcZdovc) [inline components/brave_rewards/extension/brave_rewards/components/panel.tsx]: let's use the same logic that we use for contribute table where we have support for regular sites, youtube and twitch.

https://github.com/brave/brave-core/blob/master/components/brave_rewards/ui/components/contributeBox.tsx#L49

Comment (NejcZdovc) [reply components/brave_rewards/extension/brave_rewards/components/panel.tsx]: note that we currently have only favicon for twitch VOD's and not for live videos.

Comment (jasonrsadler) [reply components/brave_rewards/extension/brave_rewards/components/panel.tsx]: Fixed

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 2]
```diff
@@ -347,15 +348,26 @@ PublisherInfoDatabase::GetMediaPublisherInfo(const std::string& media_key) {
...
-      db_.GetUniqueStatement("SELECT pi.publisher_id, pi.name, pi.url, "
-                             "pi.favIcon, pi.provider, pi.excluded, ai.percent "
-                             "FROM media_publisher_info as mpi "
-                             "INNER JOIN publisher_info AS pi ON "
-                                "mpi.publisher_id = pi.publisher_id "
-                             "INNER JOIN activity_info AS ai ON "
-                                "mpi.publisher_id = ai.publisher_id "
-                             "WHERE mpi.media_key=?"));
+  std::string sql_str;
+  if (use_as_publisher_key) {
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: This is part of regular publisher load, so we should just call that function if we need to

Comment (jasonrsadler) [reply components/brave_rewards/browser/publisher_info_database.cc]: Nothing should change from regular publisher load as `use_media_key` is an optional parameter. Should it still be split out to separate?

**Comment by jasonrsadler** [comment]
Created follow up for youtube custom path publishers: https://github.com/brave/brave-browser/issues/1886

**Comment by NejcZdovc** [comment]
master 7f686e1
0.57.x f4fa67ec
0.56.x 2dcd0e7c

**Pull request merged**